CMakeLists.txt

```diff
@@ -0,0 +1,168 @@
+# Copyright (C) 2022-2023 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+cmake_minimum_required(VERSION 3.14.0 FATAL_ERROR)
+project(unified-memory-framework VERSION 0.8.0)
+
+include(GNUInstallDirs)
+include(CheckCXXSourceCompiles)
+include(CMakePackageConfigHelpers)
+include(CTest)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+include(helpers)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED YES)
+
+# Build Options
```

Comment:
OK, I have removed all but those two: `UMF_DEVELOPER_MODE` and `UMF_BUILD_SHARED_LIBRARY` - IMHO they are very useful - what do you think?

Reply:
If possible, I would keep `UMF_BUILD_SHARED_LIBRARY` env since it is our target for the future and it might be useful for experiments even until it is not officially supported.